### PR TITLE
Add more stubs for redis-py

### PR DESCRIPTION
**Pull request by brainix**
I've tested these stubs against my own project here:
https://github.com/brainix/pottery/pull/240

The output is clean for all these new stubs when running `$ mypy
--strict`.

### CODE REVIEW

File: third_party/2and3/redis/client.pyi [hunk 1]
```diff
@@ -221,7 +221,7 @@ class Redis(object):
     def smembers(self, name: _Key) -> Set[_Value]: ...
     def smove(self, src, dst, value): ...
     def spop(self, name, count: Optional[int] = ...): ...
-    def srandmember(self, name, number=...): ...
+    def srandmember(self, name: _Key, number: Optional[int] = ...) -> _Value: ...
```
Comment (srittau) [inline third_party/2and3/redis/client.pyi:224]: This returns `Optional[_Value]` if `number` is not given or is `None` and a list of values if `number` is not `None`. We should use overloads to reflect this.

File: third_party/2and3/redis/client.pyi [hunk 2]
```diff
@@ -159,12 +159,12 @@ class Redis(object):
     ) -> Optional[bool]: ...
     def __setitem__(self, name, value): ...
     def setbit(self, name: _Key, offset: int, value: int) -> int: ...
-    def setex(self, name, time, value): ...
+    def setex(self, name: _Key, time: Union[int, timedelta], value: _Value): ...
```
Comment (srittau) [inline third_party/2and3/redis/client.pyi:162]: Returns `bool`.

**Review by srittau — CHANGES_REQUESTED** [review]
Thanks, two remarks below.

**Comment by brainix** [comment]
Hi, @srittau.

Thanks for the feedback; I believe I've addressed your concerns. However, now, whenever I run `$ mypy --strict`, one of the errors that I get is:

**Comment by brainix** [comment]
Thanks for the feedback, @srittau. I believe that I've addressed your concerns. How does this Pull Request look now?

File: third_party/2and3/redis/client.pyi [hunk 3]
```diff
@@ -221,7 +221,10 @@ class Redis(object):
     def smembers(self, name: _Key) -> Set[_Value]: ...
     def smove(self, src, dst, value): ...
     def spop(self, name, count: Optional[int] = ...): ...
-    def srandmember(self, name, number=...): ...
+    @overload
+    def srandmember(self, name: _Key, number: None) -> Optional[_Value]: ...
+    @overload
+    def srandmember(self, name: _Key, number: int) -> List[_Value]: ...
```
Comment (srittau) [inline third_party/2and3/redis/client.pyi:227]: One little thing and then this is ready to go. The first overload needs `number: None = ...` as `srandmember()` has a default (`None`) for the second argument.

**Review by srittau — APPROVED** [review]

**Pull request merged**